tidy3d/components/source.py

```diff
@@ -463,6 +463,45 @@ def _dir_vector(self) -> Tuple[float, float, float]:
         return None
 
 
+class BroadbandSource(Source, ABC):
+    """A source with frequency dependent field distributions."""
+
+    num_freqs: int = pydantic.Field(
+        1,
+        title="Number of Frequency Points",
+        description="Number of points used to approximate the frequency dependence of injected "
+        "field. A Chebyshev interpolation is used, thus, only a small number of points, i.e., less "
+        "than 20, is typically sufficient to obtain converged results.",
+        ge=1,
+        le=99,
+    )
+
+    @cached_property
+    def frequency_grid(self) -> np.ndarray:
+        """A Chebyshev grid used to approximate frequency dependence."""
+        freq_min, freq_max = self.source_time.frequency_range(num_fwidth=4)
+        freq_avg = 0.5 * (freq_min + freq_max)
+        freq_diff = 0.5 * (freq_max - freq_min)
+        uni_points = (2 * np.arange(self.num_freqs) + 1) / (2 * self.num_freqs)
+        cheb_points = np.cos(np.pi * np.flip(uni_points))
+        return freq_avg + freq_diff * cheb_points
+
+    @pydantic.validator("num_freqs", always=True, allow_reuse=True)
+    def _warn_if_large_number_of_freqs(cls, val):
+        """Warn if a large number of frequency points is requested."""
+
+        if val is None:
+            return val
+
+        if val >= 20:
+            logging.warning(
+                f"A large number ({val}) of frequency points is used in a broadband source. "
+                "It is expected that less than 20 points is typically sufficient."
```

Comment:
Also nitpicky, but I suggest slightly different wording, something like "A large number ({val}) of frequency points is used in a broadband source. This can slow down simulation time and is only needed if the mode fields are expected to have a very sharp frequency dependence. 'num_freqs' < 20 is sufficient in most cases."